### DISCUSSION TITLE
with #367 How can I get the page count of each page externally

### DISCUSSION THREAD

**Discussion by nfines** [QuestPDF/QuestPDF#587] [Q&A] [answered]
Document doc=Document
    .Create(document =>
    {
       **//pageA(assume total 3 pages)**
        document.Page(page =>
        {
            page.Content().Text("Content for user A");
            page.Footer().Text(text => text.CurrentPageNumber());
        });
        **//pageB(assume total 5 pages)**
        document.Page(page =>
        {
            page.Content().Text("Content for user B");
            page.Footer().Text(text => text.CurrentPageNumber());
        });
        **//pageC(assume total 7 pages)**
        document.Page(page =>
        {
            page.Content().Text("Content for user C");
            page.Footer().Text(text => text.CurrentPageNumber());
        });
    });

**How to know the total number of page A,B,C**
int pageA_totalNumber=doc.functionToGetTotalPage("PageA"); //like this got 3
int pageB_totalNumber=doc.functionToGetTotalPage("PageB"); //like this got 5
int pageC_totalNumber=doc.functionToGetTotalPage("PageC"); //like this got 7

**Comment by MarcinZiabek** [comment]
Please take a look at the following APIs:
- [Section](https://www.questpdf.com/api-reference/section.html) element,
- [SectionLink](https://www.questpdf.com/api-reference/section-link.html) element,
- support for sections in the [Text](https://www.questpdf.com/api-reference/text.html#page-numbers) element.

I hope it will be sufficient 😁

**Reply by nfines** [reply]
thanks for reply first.
Because I want to switch from some buttons to specified pages, I must know which page number are those pages.
Without knowing the exact number of pages, I at least know the number of pages in each chapter to convert the page number
now. i use PdfiumViewer to load questPDF ,then got total pages,...

**Reply by MarcinZiabek** [reply, marked as answer]
Currently, the library does not provide such details after the document generation. If more people show interest in this feature, I am open to introducing it. Are you aware of any other library that has similar capabilities?

Let me suggest a workaround. You can use [the Dynamic component](https://www.questpdf.com/concepts/dynamic-components.html#introduction) that renders empty content and place it at the very beginning of your sections. That component provides the current page number, which you can cache and save in some global (or injected) object.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
